### DISCUSSION TITLE
01. PyTorch Workflow Fundamentals Timestand - 5:34:45sec

### DISCUSSION THREAD

**Discussion by Gauravjake** [mrdbourke/pytorch-deep-learning#234] [Q&A]
here are the code and I  got  the error that object has no attribute 'weights'

from torch import nn

# Create lines repression model class
class LinearRegressionModel(nn.Module): # -> almost everything in Pytorch inherhits from nn.Module
  def _init_(self):
   super()._init_()
   self.weights = nn.Parameter(torch.randn(1, # <- start with a random weight and try to adjust it to the ideal weight
                                          requires_grad=True, # <- can this parameter be updated via gradient descent?
                                          dtype=torch.float)) # <- Pytorch loves the datatype torch float32

   self.bias = nn.Parameter(torch.randn(1, #<- start with a  random bias and try to  adjust it to the ideal bias 
                                       require_grad=True, # <- can this parameter be update via gradient descent?
                                       dtype=torch.float)) # <- Python loves the datatype torch.float32

# Forward method to define the computation is the node:
  def forward(self, x: torch.Tensor) -> torch.Tensor: # <- "x" is the input data
   return self.weights * x + self.bias # this is  the linear regression formula
``

AttributeError                            Traceback (most recent call last)
[<ipython-input-88-8c71e82041ef>](https://localhost:8080/#) in <module>
      2 with torch.inference_mode():
      3 
----> 4   y_preds = model_0(x_test)
      5 

2 frames
[/usr/local/lib/python3.8/dist-packages/torch/nn/modules/module.py](https://localhost:8080/#) in __getattr__(self, name)
   1263             if name in modules:
   1264                 return modules[name]
-> 1265         raise AttributeError("'{}' object has no attribute '{}'".format(
   1266             type(self).__name__, name))
   1267 

AttributeError: 'LinearRegressionModel' object has no attribute 'weights'

.

>   **** Can any please explain me what is going on here. I search  this in stack overflow but still  can fix my problem and doubt.

**Comment by BoolinBood** [comment]
on line 3, try changing super().init() to super().__init__(), I don't know this might fix the problem but let give it a try.

**Comment by PatTheAtak** [comment]
Hey @Gauravjake , I hope you're well. 
So, yeah, I've gone through the piece of code you've posted and I think you've missed some OOP syntax when calling the super class' constructor. 
Try this piece of code and see if it works.

```
class LinearRegressionModel(nn.Module):
    def __init__(self):
        super().__init__() 
        self.weights = nn.Parameter(torch.randn(1,
            requires_grad=True, 
            dtype=torch.float 
        ))
        self.bias = nn.Parameter(torch.randn(1, 
            requires_grad=True, 
            dtype=torch.float 
        ))
```
